### PR TITLE
Fix: Output was being overwritten when there were multiple Node->x connections

### PR DESCRIPTION
**Pull request by mundya**
I believe this is a fix for #70.

@mossblaser can you give it a go and let me know what happens?  Thank you!

### CODE REVIEW

File: nengo_spinnaker/nodes/ethernet.py [hunk 1]
```diff
@@ -178,7 +178,8 @@ def setup(self, machinename, port, input_period, nodes_tx,
 
                 # Store the Rx, Connection and Buffer slice
                 i = crx.rx.connections[crx.connection]
-                _transform = crx.rx.connections.transforms_functions[i].transform
+                _transform = \
+                    crx.rx.connections.transforms_functions[i].transform
```
Comment (mundya) [inline nengo_spinnaker/nodes/ethernet.py:182]: `flake8`

File: nengo_spinnaker/nodes/ethernet.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ def get_node_out_vertex(self, builder, conn):
         rx = None
         for _rx in self.nodes_rxes[conn.pre]:
             # See if we've already assigned a similar connection for this Node
-            if rx.contains_compatible_connection(conn):
+            if _rx.contains_compatible_connection(conn):
```
Comment (mundya) [inline nengo_spinnaker/nodes/ethernet.py:88]: `rx` may have been None, `_rx` is the element of `self.nodes_rxes`.

**Comment by mossblaser** [comment]
I get:

```
Traceback (most recent call last):
  File "/tmp/simple_broken.py", line 18, in <module>
    sim = nengo_spinnaker.Simulator(model)
  File "/home/jonathan/Programing/spinnaker/nengo_summer_school/nengo_spinnaker/nengo_spinnaker/simulator.py", line 61, in __init__
    self.builder(model, dt, seed, node_builder=io, config=config)
  File "/home/jonathan/Programing/spinnaker/nengo_summer_school/nengo_spinnaker/nengo_spinnaker/builder.py", line 135, in __call__
    self._build(conn)
  File "/home/jonathan/Programing/spinnaker/nengo_summer_school/nengo_spinnaker/nengo_spinnaker/builder.py", line 67, in _build
    self.builders[obj_class](obj)
  File "/home/jonathan/Programing/spinnaker/nengo_summer_school/nengo_spinnaker/nengo_spinnaker/builder.py", line 202, in _build_connection
    edge = edge_builders[key](self, c)
  File "/home/jonathan/Programing/spinnaker/nengo_summer_school/nengo_spinnaker/nengo_spinnaker/utils/probes.py", line 63, in build_node_probenode_edge
    prevertex = builder.get_node_out_vertex(c)
  File "/home/jonathan/Programing/spinnaker/nengo_summer_school/nengo_spinnaker/nengo_spinnaker/builder.py", line 232, in get_node_out_vertex
    return self.node_builder.get_node_out_vertex(self, c)
  File "/home/jonathan/Programing/spinnaker/nengo_summer_school/nengo_spinnaker/nengo_spinnaker/nodes/ethernet.py", line 98, in get_node_out_vertex
    if _rx.n_remaining_dimensions >= conn.size_out:
AttributeError: 'Connection' object has no attribute 'size_out'
```

When running this. This can be fixed by changing:

```
diff --git a/nengo_spinnaker/nodes/ethernet.py b/nengo_spinnaker/nodes/ethernet.py
index 79d85cd..1530b34 100644
--- a/nengo_spinnaker/nodes/ethernet.py
+++ b/nengo_spinnaker/nodes/ethernet.py
@@ -95,7 +95,7 @@ class Ethernet(object):
             # See if we have space for this connection in any of the existing
             # Rxes for this Node
             for _rx in self.nodes_rxes[conn.pre]:
-                if _rx.n_remaining_dimensions >= conn.size_out:
+                if _rx.n_remaining_dimensions >= conn.dimensions:
                     rx = _rx
                     break
             else:
```

--(I don't have commit access so I can't contribute, sorry!)-- Correction: I can and have committed this change to the branch.

**Comment by mossblaser** [comment]
This bug appears to be fixed in this branch. @mundya please can you sign off?

(Note: I have not looked for and fixed any other instances of the `size_out`/`dimensions` issue...)

**Comment by mundya** [comment]
[Update: sorry, multiple issues!]

Cool, so it wasn't a size out/dimensions issue.  The code for computing the output to be sent to the board reassigned some transformed version of the output to the same variable it was receiving.  (So if the first transform changed the dimensionality this caused the dimension mismatch for the next.)

All in all a fun bug!

File: nengo_spinnaker/nodes/ethernet.py [hunk 3]
```diff
@@ -95,7 +95,7 @@ def get_node_out_vertex(self, builder, conn):
             # See if we have space for this connection in any of the existing
             # Rxes for this Node
             for _rx in self.nodes_rxes[conn.pre]:
-                if _rx.n_remaining_dimensions >= conn.size_out:
+                if _rx.n_remaining_dimensions >= conn.dimensions:
```
Comment (mundya) [inline nengo_spinnaker/nodes/ethernet.py:98]: Thanks!

**Pull request merged**